opentimelineio/adapters/cmx_3600.py

```diff
@@ -590,21 +589,21 @@ def write_to_string(input_otio):
     for i, clip in enumerate(track):
         source_tc_in = otio.opentime.to_timecode(
             clip.source_range.start_time,
-            edl_rate
+            clip.source_range.start_time.rate
```

Comment:
If the rate of clips in your timeline varies, then you'll get an EDL with mixed rates in it, which is unlikely to work properly. It would be better to use a single edl_rate to make sure they are all consistent.

I would suggest adding an `edl_rate=None` parameter to `write_to_string` and then add some code to detect the rate of the timeline when edl_rate is None (maybe `timeline.duration().rate`). That way the caller could override it, but it would auto-detect if you don't specify a rate.